lib/src/back_end/code_writer.dart

```diff
@@ -174,9 +174,7 @@ class CodeWriter {
 
     var state = _pieceStates.pieceState(piece);
 
-    // TODO(tall): Support pieces with different split costs, and possibly
-    // different costs for each state value.
-    if (state != State.initial) _cost++;
+    _cost += state.cost;
```

Comment:
This is another impactful change worthy of mention in the changelog.

Reply:
Good point, done.